janitor/timeseries.py

```diff
@@ -0,0 +1,128 @@
+"""
+Time series-specific data testing and cleaning functions.
+"""
+
+import pandas as pd
+import pandas_flavor as pf
+from pandas.api.types import is_datetime64_any_dtype as is_datetime
+from collections import namedtuple
+from janitor import check
+
+
+def _flag_missing_timestamps(
+    df: pd.DataFrame,
+    frequency: str,
+    column_name: str,
+    first_time_stamp: pd.Timestamp,
+    last_time_stamp: pd.Timestamp,
+) -> namedtuple:
+    """
+    Test if timestamps are missing
+
+    Utility function to test if input data frame is missing any timestamps relative to expected timestamps.
+    They are generated based on the first_time_stamp, last_time_stamp and frequency.
+
+    :param df: data frame which needs to be tested for missing timestamps
+    :param frequency: frequency i.e. sampling frequency of the data.
+        A list of acceptable frequency strings are available here:
+        (https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases)
```

Comment:
Hyperlinks should follow [Sphinx-style syntax][sphinx]. This slipped from my view; thankfully, linters like Flake8 help catch this :).

[sphinx]: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#hyperlinks